utils/rpms/daos.spec

```diff
@@ -207,12 +207,11 @@ packages
 Summary: The DAOS test suite
 Requires: %{name}-client%{?_isa} = %{version}-%{release}
 Requires: %{name}-admin%{?_isa} = %{version}-%{release}
-Requires: python3-distro
-Requires: python3-tabulate
-Requires: python3-defusedxml
+Requires: %{name}-devel%{?_isa} = %{version}-%{release}
 Requires: protobuf-c-devel
 Requires: fio
 Requires: git
+Requires: gdb
```

Comment:
What test(s) requires `gdb`?

Reply:
The harness uses it if there's a core dump.  Without this change then launch.py bombs out after test failure.

Reply:
The harness installs it if needed as it's only needed in the case of there being a core dump.

> Without this change then launch.py bombs out after test failure.

Any test failure or a core dumping test failure?  The former would be very odd and the latter would be a bug that needs addressing.

Reply:
I saw this with the daos_build test which always generates a core dump from "conftest", and that core dump it whitelisted from causing a test failure - but it seems that gdb is still used to generate a stack trace from it.  The point was that gdb being missing was a hard failure for launch.py and not handled.

Previously the CI post node install script was using dnf to install avocado-framework rpms which have a dependency on gdb, with that gone from the script then gdb was not installed and ftest/launch.py was aborting.

Maybe something to fix so that gdb itself is installed on demand in the same way as dependencies are when a core dump is detected but adding gdb to the rpm dependency seemed the best option as there is already a requirement for it for the code in the rpm to run.

Reply:
Here's the failing run from where gdb was missing

https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13565/38/testReport/junit/FTEST_launch/dfuse-daos_build/01___dfuse_daos_build_py/

Reply:
Here is why `gdb` was not found:

```
2024/01/08 03:55:42 DEBUG                      run_local: Running on wolf-102vm1: sudo dnf -y --enablerepo=*debug* install gdb systemd-debuginfo-239-74.el8_8 ndctl-debuginfo-71.1-4.el8 mercury-debuginfo-2.3.1-2.el8 hdf5-debuginfo-1.13.1-3.el8 argobots-debuginfo-1.1-3.el8 libfabric-debuginfo-1.19.0-1.el8 hdf5-vol-daos-mpich-debuginfo-1.1.0-3.el8 hdf5-vol-daos-mpich-tests-debuginfo-1.1.0-3.el8 hdf5-vol-daos-openmpi-debuginfo-1.1.0-3.el8 hdf5-vol-daos-openmpi-tests-debuginfo-1.1.0-3.el8 ior-debuginfo-3.3.0-20.gd3574d5.el8
2024/01/08 03:55:44 DEBUG                      run_local:   wolf-102vm1 (rc=1):
2024/01/08 03:55:44 DEBUG                      run_local:     created by dnf config-manager from https://buil  84 kB/s | 3.0 kB     00:00    
2024/01/08 03:55:44 DEBUG                      run_local:     daos_ci-rocky8-base-artifactory                 228 kB/s | 4.3 kB     00:00    
2024/01/08 03:55:44 DEBUG                      run_local:     daos_ci-rocky8-extras-artifactory                78 kB/s | 1.4 kB     00:00    
2024/01/08 03:55:44 DEBUG                      run_local:     daos_ci-rocky8-appstream-artifactory            260 kB/s | 4.8 kB     00:00    
2024/01/08 03:55:44 DEBUG                      run_local:     daos_ci-rocky8-powertools-artifactory           250 kB/s | 4.8 kB     00:00    
2024/01/08 03:55:44 DEBUG                      run_local:     daos_ci-rocky8-base-artifactory-debuginfo        81 kB/s | 1.4 kB     00:00    
2024/01/08 03:55:44 DEBUG                      run_local:     daos_ci-rocky8-extras-artifactory-debuginfo      77 kB/s | 1.4 kB     00:00    
2024/01/08 03:55:44 DEBUG                      run_local:     daos_ci-rocky8-appstream-artifactory-debuginfo  166 kB/s | 3.1 kB     00:00    
2024/01/08 03:55:44 DEBUG                      run_local:     daos_ci-rocky8-powertools-artifactory-debuginfo 164 kB/s | 3.1 kB     00:00    
2024/01/08 03:55:44 DEBUG                      run_local:     daos_ci-el8-epel-artifactory                    280 kB/s | 4.7 kB     00:00    
2024/01/08 03:55:44 DEBUG                      run_local:     daos_ci-el8-epel-modular-artifactory            174 kB/s | 3.0 kB     00:00    
2024/01/08 03:55:44 DEBUG                      run_local:     daos-stack-deps-el-8-x86_64-stable-local-artifa 9.9 kB/s | 475  B     00:00    
2024/01/08 03:55:44 DEBUG                      run_local:     daos_ci-intel-tools-el-8-x86_64-stable-local-ar 9.6 kB/s | 475  B     00:00    
2024/01/08 03:55:44 DEBUG                      run_local:     No match for argument: systemd-debuginfo-239-74.el8_8
2024/01/08 03:55:44 DEBUG                      run_local:     Error: Unable to find a match: systemd-debuginfo-239-74.el8_8
```

So that's a bug in the _debuginfo_ package(s) installation.  The solution is to fix that, not add `gdb` to this package.

Reply:
I'm not sure I follow, is this a bug in our packaging or somewhere else.  I have no problem adding a fix to this PR but I don't understand enough about the debuginfo process or our provisioning to fix this.

In the short term I could revert the changes related to gdb in this PR so remove it from the rpm dependency again and have the provisioning script install it so our code would be "no worse than before" but I don't know what's needed to fix it properly.

Reply:
> I'm not sure I follow, is this a bug in our packaging or somewhere else.

It's a bug in the code in the function test framework that installs `gdb` and the slew of `-debuginfo` packages when core files are found.

I would prefer to keep this method rather than adding yet more, not-really-necessary dependencies to the daos packaging.

>  I have no problem adding a fix to this PR but I don't understand enough about the debuginfo process or our provisioning to fix this.

Perhaps the best thing to do for now is just remove this dependency and open a ticket to fix the root issue.
 
> In the short term I could revert the changes related to gdb in this PR so remove it from the rpm dependency again and have the provisioning script install it so our code would be "no worse than before" but I don't know what's needed to fix it properly.

This would be at least somewhat better than adding a new unnecessary dependency.

Reply:
I think the thing you're missing here Brian is that if I remove this dependency from the PR and gdb isn't installed in some other way then the testing will fail and not give stack trace results.  Yes I'll file a bug that this is broken, it happens to work today because the avocado rpms which are installed in the `ci/provisioning/post_provision_config_nodes_EL_8.sh` script depend on gdb, this PR does not install the avocado packages in that script any more so gdb is not installed, without some other mechanism to install them then testing will fail, not just with test failures but with a framework error and without giving the valuable stack-trace contents etc.

In order to not break ftest and cause all testing runs to fail then this PR needs a solution, the approaches I can think of are:
1. Add gdb to the rpm dependencies
2. Install gdb explicitely in ci/provisioning/post_provision_config_nodes_EL_8.sh
3. Mark this PR as blocked on the as-yet-unfiled debuginfo bug.

Given that all test runs will likely require gdb my preferred solution is 1. however 2. is closest to the existing process except that it's currently installed indirectly rather than directly.

Reply:
I just tried the failing command from the link above, `dnf -y --enablerepo=*debug* install gdb systemd-debuginfo-239-74.el8_8 ndctl-debuginfo-71.1-4.el8 mercury-debuginfo-2.3.1-2.el8 hdf5-debuginfo-1.13.1-3.el8 argobots-debuginfo-1.1-3.el8 libfabric-debuginfo-1.19.0-1.el8 hdf5-vol-daos-mpich-debuginfo-1.1.0-3.el8 hdf5-vol-daos-mpich-tests-debuginfo-1.1.0-3.el8 hdf5-vol-daos-openmpi-debuginfo-1.1.0-3.el8 hdf5-vol-daos-openmpi-tests-debuginfo-1.1.0-3.el8 ior-debuginfo-3.3.0-20.gd3574d5.el8` on a active test node and it passed this time, perhaps this issue has been fixed anyway through a change in available distro packages?